### ISSUE TITLE
Size mismatch for embeddingnet.masks.weight

### ISSUE THREAD

**Issue by Karenou** [mvasil/fashion-compatibility#31]
Hi, I have a question about the number of type spaces in the pre-trained model.

As in the main.py, the default num_rand_embed is 4. But when I load the pre-trained model, it shows that the shape of embeddingnet.masks.weight is [66,64], while my current tnet is [4, 64]. May I know if the num_rand_embed in the pretrained model is 66?

Many thanks!

**Comment by AliHentati2** [comment]
add -> keepdim=True <- in norm = torch.norm(masked_embedding, p=2, dim=2) + 1e-10 so that it becomes norm = torch.norm(masked_embedding, p=2, dim=2, keepdim=True) + 1e-10

**Comment by marzooq-unbxd** [comment]
But using this , 
```
sample_images = next(iter(test_loader)).to(device)
sample_embeddings = tnet.embeddingnet(sample_images)
```
I get sample_embeddings of shape [256, 67, 64] if sample_images is of [256, 3, 112, 112]
Do you get the same shape? (emphasis on the 67)

Also I get  len(test_loader.dataset.typespaces) as 66, 
keepdim=True makes it 67?
